modules/src/clients/ics13_near/client_def.rs

```diff
@@ -307,14 +308,15 @@ pub fn validate_light_block<H: HostFunctionsProvider>(
 
         approved_stake += bp_stake;
 
-        let _validator_public_key = bp_stake_view.public_key.clone();
-        // if !maybe_signature
-        //     .as_ref()
-        //     .unwrap()
-        //     .verify::<H>(&H::sha256_digest(&approval_message), validator_public_key.clone())
-        // {
-        //     return Err(NearError::invalid_signature().into());
-        // }
+        let validator_public_key = &bp_stake_view.public_key;
+        let data = H::sha256_digest(&approval_message);
+        if !maybe_signature
+            .as_ref()
+            .unwrap()
+            .verify::<H>(&data, validator_public_key)
```

Comment:
We could make this a host function?

Reply:
I have no problems with that. But let's figure out first https://github.com/ComposableFi/ibc-rs/pull/8#discussion_r883471726

Reply:
done in [93d3cb5](https://github.com/ComposableFi/ibc-rs/pull/8/commits/93d3cb5835b48ab3588e899e0e49bacb70a2e853)